### PR TITLE
Extract OpenAI API retry handler and unify ADA embeddings calls.

### PR DESCRIPTION
**Pull request by collijk**
### Background
This is some leftover work from yesterday where I started to unify the ada embedding code.  It extracts a decorator for retrying the openai API (to be applied to create_chat_completion in another PR, OOS for this one, but is on my critical path for removing global state).  It then creates a single function with robust retries that is used by everyone to get embeddings.

### Changes
- Extract a retry decorator for querying the openai api
- Removed the embedding code from the api manager
- Unify the functions to retrieve ada embeddings

### Documentation
- New docstrings added.

### Test Plan
Added test suite for extracted retry handler

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/api_manager.py [hunk 1]
```diff
@@ -65,32 +65,6 @@ def create_chat_completion(
         self.update_cost(prompt_tokens, completion_tokens, model)
         return response
 
-    def embedding_create(
```
Comment (collijk) [inline autogpt/api_manager.py:68]: This class has too much scope and insufficiently encapsulates existing code.  Next PR will leave it well defined around resource management only.

File: autogpt/llm_utils.py [hunk 2]
```diff
@@ -13,10 +14,61 @@
 from autogpt.types.openai import Message
 
 CFG = Config()
-
 openai.api_key = CFG.openai_api_key
 
 
+def retry_openai_api(
```
Comment (collijk) [inline autogpt/llm_utils.py:20]: To apply to `call_ai_function`  as well in another PR.  OOS for this one.

File: tests/test_api_manager.py [hunk 3]
```diff
@@ -86,37 +86,6 @@ def test_create_chat_completion_valid_inputs():
             assert api_manager.get_total_completion_tokens() == 20
             assert api_manager.get_total_cost() == (10 * 0.002 + 20 * 0.002) / 1000
 
-    @staticmethod
```
Comment (collijk) [inline tests/test_api_manager.py:89]: This method no longer present on api_manager

File: autogpt/api_manager.py (hunk 1 above)
Comment (collijk) [reply autogpt/api_manager.py:68]: And will be renamed as `ResourceManager`

**Review by merwanehamadi — APPROVED** [review]

**Pull request merged**